feat(colors): update colors to match V2 palette

aledavila opened the pull request:
Closes #3737 

Updates color elements package to V2 color palette

#### Changelog

**New**

- yellow30 alert color

**Changed**

- all colors

Commit:
chore(colors): change yellow 20 to 30

Commit:
chore(colors): update color palette to v2

Commit:
chore(colors): deprecate yellow 20

Commit:
chore(theme): remove from depreaction as its not in theme

joshblack commented:
@aledavila I think this should be a `feat` given the deprecation and value changes 👍

joshblack reviewed (changes requested):
Could you move `yellow20` to a deprecated comment block at the end of the file? Removing it would be considered a breaking change 👍

joshblack commented on packages/colors/src/colors.js:
```diff
@@ -268,6 +268,6 @@ export const colors = {
   },
   yellow: {
-    20: yellow20,
+    30: yellow30,
```
I think we'll also need to keep yellow20 in here, as well, technically

aledavila replied:
Should I leave yellow as yellow 30 and add yellow 20 as its own ?

joshblack replied:
I think adding them both under yellow is fine 👍 This is to prevent code from breaking when it uses `colors.yellow['20']` 👍

aledavila replied:
Done

Commit:
chore(colors): deprecate yellow

shixiedesign commented on packages/colors/src/colors.js:
```diff
@@ -11,22 +11,22 @@ export const black100 = black;
-export const red50 = '#fb4b53';
+export const red10 = '#fff1f1';
+export const red20 = '#ffd7d9';
+export const red30 = '#ffb2b8';
```
This one is should be `#ffb3b8`

shixiedesign commented on packages/colors/src/colors.js:
```diff
@@ -63,16 +63,16 @@ export const magenta = {
+export const purple30 = '#d4bbff';
+export const purple40 = '#be95ff';
+export const purple50 = '#a56eff';
+export const purple60 = '#8c3ffc';
```
Spotted! Should be `#8A3FFC`

shixiedesign commented on packages/colors/src/colors.js:
```diff
@@ -132,15 +132,15 @@ export const cyan = {
-export const teal90 = '#003137';
+export const teal70 = '#005d5d';
+export const teal80 = '#004144';
+export const teal90 = '#002b30';
```
should be `#022b30`

shixiedesign reviewed (commented):
Thanks for taking this on! Found only 3 colors hex wrong, everything else is correct! 👍

Commit:
chore(colors): fix some colors

joshblack commented:
Looks great! Only comment would be making sure the commit type is `feat` when merging, maybe something like:

```
feat(colors): update colors to match v2 palette
```

joshblack commented:
@aledavila ~~seems like there is an issue with the build step that's causing CI to fail~~ 🤔 

Just pushed up a fix 👍

joshblack committed:
fix(colors): update build task and move block

The deprecated block was defined after colors which caused the value of
yellow20 to be undefined. In turn, the build process failed because the
value for that color was undefined.

joshblack commented:
Looks like all that's left is updating the snapshot 👍

Commit:
chore(colors): update snapshot

Commit:
chore(colors): update snapshot after pulling

asudoh commented on packages/colors/src/colors.js:
```diff
@@ -247,6 +247,9 @@ export const warmGray = {
 };
 
+// Deprecated ☠️
+export const yellow20 = '#fdd13a';
```
Do you want to defer this removal until next major?

aledavila replied:
Should we remove the deprecation notice ? I'm unsure otherwise how to add both colors since yellow 30 is part of the updated palette. unless we want to leave the name as yellow20 with the new hex @shixiedesign ?

joshblack replied:
This code should be safe, since there is only one "yellow" we should expose `yellow30` as yellow. `yellow20` is for backwards compat 👍

asudoh replied:
I thought the lines were removed instead of added. Sorry for the confusion.

shixiedesign replied:
Yeah the new yellow is `yellow30`. Sounds like you guys worked out how to handle this part.

asudoh reviewed (approved):
LGTM 👍, just one thing - Thanks @aledavila!

Commit:
Merge branch 'master' into v2-palette-update

Commit:
Merge branch 'master' into v2-palette-update

shixiedesign reviewed (approved):
Thank you! Really appreciate this work!

chrisconnors-ibm commented:
How did reviewers determine that the changes were complete (all desired changes made) and correct (all changes have correct values)?

joshblack committed:
Merge branch 'master' into v2-palette-update

joshblack commented:
Ended up not checking the values themselves, but making sure they were implemented correctly (made sure deprecations were made and exports are consistent).

Posted this in the linked comment, but for future reference we have this file to show what has changed value-wise: https://github.com/carbon-design-system/carbon/pull/3957/files#diff-cfac639f8d5484e0541973e7fdb5cca0 can't confirm the exact value though, on my end.

shixiedesign committed:
Merge branch 'master' into v2-palette-update

shixiedesign commented:
@chrisconnors-ibm I went over each one of the 104 swatches to check with Sketchfile. I understand ideally we'd have a text file for dev to copy the hex code over but it's just too much work this sprint. Thank you Ale for putting up with the busy work. This is great and I think having 2 pairs of eyes help with making sure the values are correct!
